src/Lib/Canvas.vala

```diff
@@ -87,6 +87,22 @@ public class Akira.Lib.Canvas : Goo.Canvas {
         window.event_bus.set_focus_on_canvas.connect (on_set_focus_on_canvas);
     }
 
+    public void insert_item_default (Akira.Lib.Models.CanvasItem item, bool select = false) {
+        items_manager.add_item (item);
+        // TODO: place in the viewport
+        Utils.AffineTransform.set_position (
+            Akira.Layouts.MainCanvas.CANVAS_SIZE / 2,
+            Akira.Layouts.MainCanvas.CANVAS_SIZE / 2,
+            item    
+        );
```

Comment:
We could implement a check to see if an item is currently selected, and use its X & Y coordinates to position the image, and if not, put it in the middle of the canvas like you did.